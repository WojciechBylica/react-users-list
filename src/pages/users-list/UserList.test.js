import React from 'react';
import { shallow } from 'enzyme';
import UsersList from '.';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';

const users = [
  {
    name: {
      first: 'Dominic',
      last: 'Onderwater',
    },
    location: {
      street: {
        number: 7174,
        name: 'Bontekamp',
      },
      city: 'Pieterburen',
      coordinates: {
        latitude: '44.5991',
        longitude: '-37.1930',
      },
    },
    email: 'dominic.onderwater@example.com',
    login: {
      uuid: '387b2ae0-2637-4599-b725-abb168c1241f',
    },
    registered: {
      date: '2014-12-11T04:02:22.729Z',
    },
    picture: {
      thumbnail: 'https://randomuser.me/api/portraits/thumb/men/6.jpg',
    },
  },
  {
    name: {
      first: 'Rosalyn',
      last: 'Steward',
    },
    location: {
      street: {
        number: 4496,
        name: 'Dublin Road',
      },
      city: 'Waterford',
      coordinates: {
        latitude: '-37.7173',
        longitude: '1.0768',
      },
    },
    email: 'rosalyn.steward@example.com',
    login: {
      uuid: 'ddc905ff-8cc6-4141-895a-fcebd72fe652',
    },
    registered: {
      date: '2018-06-30T16:34:15.754Z',
    },
    picture: {
      thumbnail: 'https://randomuser.me/api/portraits/thumb/women/48.jpg',
    },
  },
];
//jest snapshots
describe('should render ok', () => {
  it('should render list of 2 users', () => {
    const tree = renderer.create(
      <BrowserRouter>
        <UsersList
          users={users}
          loading={false}
          hasError={false}
          resultsNumber={2}
        />
      </BrowserRouter>
    );
    expect(tree).toMatchSnapshot();
  });

  it('should render error if empty user array provided', () => {
    const tree = renderer.create(
      <BrowserRouter>
        <UsersList
          users={[]}
          loading={false}
          hasError={false}
          resultsNumber={2}
        />
      </BrowserRouter>
    );
    expect(tree).toMatchSnapshot();
  });

  it('should render "loading" and 2 placeholders if loading is true', () => {
    const tree = renderer.create(
      <BrowserRouter>
        <UsersList
          users={users}
          loading={true}
          hasError={false}
          resultsNumber={2}
        />
      </BrowserRouter>
    );
    expect(tree).toMatchSnapshot();
  });
  it('should render error if no props provided', () => {
    const tree = renderer.create(
      <BrowserRouter>
        <UsersList />
      </BrowserRouter>
    );
    expect(tree).toMatchSnapshot();
  });
});

//eznyme
describe('UserList component', () => {
  it('should render 2 components User and no error message, if users array length is 2. Should render no User components and error message if user array is empty', () => {
    const usersList = shallow(
      <UsersList
        users={[]}
        loading={false}
        hasError={false}
        resultsNumber={2}
      />
    );
    expect(usersList.find('[data-test="user"]').length).toBe(0);
    expect(usersList.find('[data-test="usersList--error"]').length).toBe(1);
    usersList.setProps({ users: users });
    expect(usersList.find('[data-test="user"]').length).toBe(2);
    expect(usersList.find('[data-test="usersList--error"]').length).toBe(0);
  });
  it('should render "loading" if loading is true', () => {
    const usersList = shallow(
      <UsersList
        users={users}
        loading={true}
        hasError={false}
        resultsNumber={2}
      />
    );

    let loader = usersList.find('[data-test="usersList--loader"]');
    expect(loader.length).toBe(1);
    usersList.setProps({ loading: false });
    loader = usersList.find('[data-test="usersList--loader"]');
    expect(loader.length).toBe(0);
  });
});
