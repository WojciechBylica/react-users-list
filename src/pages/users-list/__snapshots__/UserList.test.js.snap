// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ok should render "loading" and 2 placeholders if loading is true 1`] = `
Array [
  <p
    className="loader"
  >
    loading...
  </p>,
  <div>
    <div
      className="userWrapper"
    >
      <div
        className="media-block"
        style={
          Object {
            "display": "flex",
          }
        }
      >
        <div
          className="round-shape"
          style={
            Object {
              "backgroundColor": "#E0E0E0",
              "borderRadius": "500rem",
              "height": "100%",
              "marginRight": 10,
              "minHeight": 55,
              "minWidth": 55,
              "width": 55,
            }
          }
        />
        <div
          className="text-block"
          style={
            Object {
              "width": "100%",
            }
          }
        >
          <div
            className="text-row"
            style={
              Object {
                "backgroundColor": "#E0E0E0",
                "height": "1em",
                "marginTop": 0,
                "maxHeight": "9.090909090909092%",
                "width": "97%",
              }
            }
          />
          <div
            className="text-row"
            style={
              Object {
                "backgroundColor": "#E0E0E0",
                "height": "1em",
                "marginTop": "0.7em",
                "maxHeight": "9.090909090909092%",
                "width": "100%",
              }
            }
          />
          <div
            className="text-row"
            style={
              Object {
                "backgroundColor": "#E0E0E0",
                "height": "1em",
                "marginTop": "0.7em",
                "maxHeight": "9.090909090909092%",
                "width": "94%",
              }
            }
          />
          <div
            className="text-row"
            style={
              Object {
                "backgroundColor": "#E0E0E0",
                "height": "1em",
                "marginTop": "0.7em",
                "maxHeight": "9.090909090909092%",
                "width": "90%",
              }
            }
          />
          <div
            className="text-row"
            style={
              Object {
                "backgroundColor": "#E0E0E0",
                "height": "1em",
                "marginTop": "0.7em",
                "maxHeight": "9.090909090909092%",
                "width": "98%",
              }
            }
          />
          <div
            className="text-row"
            style={
              Object {
                "backgroundColor": "#E0E0E0",
                "height": "1em",
                "marginTop": "0.7em",
                "maxHeight": "9.090909090909092%",
                "width": "95%",
              }
            }
          />
        </div>
      </div>
    </div>
    <div
      className="userWrapper"
    >
      <div
        className="media-block"
        style={
          Object {
            "display": "flex",
          }
        }
      >
        <div
          className="round-shape"
          style={
            Object {
              "backgroundColor": "#E0E0E0",
              "borderRadius": "500rem",
              "height": "100%",
              "marginRight": 10,
              "minHeight": 55,
              "minWidth": 55,
              "width": 55,
            }
          }
        />
        <div
          className="text-block"
          style={
            Object {
              "width": "100%",
            }
          }
        >
          <div
            className="text-row"
            style={
              Object {
                "backgroundColor": "#E0E0E0",
                "height": "1em",
                "marginTop": 0,
                "maxHeight": "9.090909090909092%",
                "width": "97%",
              }
            }
          />
          <div
            className="text-row"
            style={
              Object {
                "backgroundColor": "#E0E0E0",
                "height": "1em",
                "marginTop": "0.7em",
                "maxHeight": "9.090909090909092%",
                "width": "100%",
              }
            }
          />
          <div
            className="text-row"
            style={
              Object {
                "backgroundColor": "#E0E0E0",
                "height": "1em",
                "marginTop": "0.7em",
                "maxHeight": "9.090909090909092%",
                "width": "94%",
              }
            }
          />
          <div
            className="text-row"
            style={
              Object {
                "backgroundColor": "#E0E0E0",
                "height": "1em",
                "marginTop": "0.7em",
                "maxHeight": "9.090909090909092%",
                "width": "90%",
              }
            }
          />
          <div
            className="text-row"
            style={
              Object {
                "backgroundColor": "#E0E0E0",
                "height": "1em",
                "marginTop": "0.7em",
                "maxHeight": "9.090909090909092%",
                "width": "98%",
              }
            }
          />
          <div
            className="text-row"
            style={
              Object {
                "backgroundColor": "#E0E0E0",
                "height": "1em",
                "marginTop": "0.7em",
                "maxHeight": "9.090909090909092%",
                "width": "95%",
              }
            }
          />
        </div>
      </div>
    </div>
  </div>,
]
`;

exports[`should render ok should render error if empty user array provided 1`] = `
<p
  className="error"
  data-test="error"
>
  an error has occurred... please check your internet connection and try again...
</p>
`;

exports[`should render ok should render error if no props provided 1`] = `
<p
  className="error"
  data-test="error"
>
  an error has occurred... please check your internet connection and try again...
</p>
`;

exports[`should render ok should render list of 2 users 1`] = `
<div>
  <a
    className="usersList__link"
    href="#/users/387b2ae0-2637-4599-b725-abb168c1241f"
    onClick={[Function]}
  >
    <div
      className="userWrapper userWrapper--link"
    >
      <h2
        className="user__header"
      >
        <img
          alt="avatar"
          className="user__avatar"
          height={48}
          src="https://randomuser.me/api/portraits/thumb/men/6.jpg"
          width={48}
        />
        <span
          className="user__person"
        >
          Dominic Onderwater
        </span>
      </h2>
      <ul
        className="user__list"
      >
        <li>
          Dominic Onderwater
        </li>
        <li>
          Bontekamp 7174 Pieterburen
        </li>
        <li>
          dominic.onderwater@example.com
        </li>
        <li>
          registered: 
          Thursday, December 11, 2014
        </li>
      </ul>
    </div>
  </a>
  <a
    className="usersList__link"
    href="#/users/ddc905ff-8cc6-4141-895a-fcebd72fe652"
    onClick={[Function]}
  >
    <div
      className="userWrapper userWrapper--link"
    >
      <h2
        className="user__header"
      >
        <img
          alt="avatar"
          className="user__avatar"
          height={48}
          src="https://randomuser.me/api/portraits/thumb/women/48.jpg"
          width={48}
        />
        <span
          className="user__person"
        >
          Rosalyn Steward
        </span>
      </h2>
      <ul
        className="user__list"
      >
        <li>
          Rosalyn Steward
        </li>
        <li>
          Dublin Road 4496 Waterford
        </li>
        <li>
          rosalyn.steward@example.com
        </li>
        <li>
          registered: 
          Saturday, June 30, 2018
        </li>
      </ul>
    </div>
  </a>
</div>
`;
